awx/main/tasks.py

```diff
@@ -87,91 +82,56 @@
+    startup_logger.info("Syncing Schedules")
+    for sch in Schedule.objects.all():
+        try:
+            sch.update_computed_fields()
+            from awx.main.signals import disable_activity_stream
+            with disable_activity_stream():
+                sch.save()
+        except Exception:
+            logger.exception(six.text_type("Failed to rebuild schedule {}.").format(sch))
 
-@celeryd_init.connect
-def celery_startup(conf=None, **kwargs):
     #
-    # When celeryd starts, if the instance cannot be found in the database,
+    # When the dispatcher starts, if the instance cannot be found in the database,
     # automatically register it.  This is mostly useful for openshift-based
     # deployments where:
     #
     # 2 Instances come online
     # Instance B encounters a network blip, Instance A notices, and
     # deprovisions it
-    # Instance B's connectivity is restored, celeryd starts, and it
+    # Instance B's connectivity is restored, the dispatcher starts, and it
     # re-registers itself
     #
     # In traditional container-less deployments, instances don't get
     # deprovisioned when they miss their heartbeat, so this code is mostly a
     # no-op.
     #
-    if kwargs['instance'].hostname != 'celery@{}'.format(settings.CLUSTER_HOST_ID):
-        error = six.text_type('celery -n {} does not match settings.CLUSTER_HOST_ID={}').format(
-            instance.hostname, settings.CLUSTER_HOST_ID
-        )
-        logger.error(error)
-        raise RuntimeError(error)
-    (changed, tower_instance) = Instance.objects.get_or_register()
-    if changed:
-        logger.info(six.text_type("Registered tower node '{}'").format(tower_instance.hostname))
+    apply_cluster_membership_policies()
+    cluster_node_heartbeat()
+    if Instance.objects.me().is_controller():
+        awx_isolated_heartbeat()
 
-    startup_logger = logging.getLogger('awx.main.tasks')
-    startup_logger.info("Syncing Schedules")
-    for sch in Schedule.objects.all():
-        try:
-            sch.update_computed_fields()
-            from awx.main.signals import disable_activity_stream
-            with disable_activity_stream():
-                sch.save()
-        except Exception:
-            logger.exception(six.text_type("Failed to rebuild schedule {}.").format(sch))
-
-    # set the queues we want to bind to dynamically at startup
-    queues = []
-    me = Instance.objects.me()
-    for q in [me.hostname] + settings.AWX_CELERY_QUEUES_STATIC:
-        q = q.encode('utf-8')
-        queues.append(Queue(q, Exchange(q), routing_key=q))
-    for q in settings.AWX_CELERY_BCAST_QUEUES_STATIC:
-        queues.append(Broadcast(q.encode('utf-8')))
-    conf.CELERY_QUEUES = list(set(queues))
-
-    # Expedite the first hearbeat run so a node comes online quickly.
-    cluster_node_heartbeat.apply([])
 
-
-@worker_shutdown.connect
-def inform_cluster_of_shutdown(*args, **kwargs):
+def inform_cluster_of_shutdown():
```

Comment:
This function gets called when the dispatcher stops.  It sets capacity to zero and reaps any running jobs assigned to the node.